CONTRIBUTING.md

```diff
@@ -1,78 +1,122 @@
+*When applicable.
 
-# How To Contribute
- - First of all [file an bug or feature report](https://github.com/flame-engine/flame/issues/new) on
- this repository.
- - [Fork the project](https://guides.github.com/activities/forking/#fork) on Github
+If the contribution doesn't meet these criteria, a maintainer will discuss it with you on the issue
+or PR. You can still continue to add more commits to the branch you have sent the Pull Request from
+and it will be automatically reflected in the PR.
+
+## Open an issue and fork the repository
+ - If it is a bigger change or a new feature, first of all
+   [file a bug or feature report][GitHub issues]
+ - [Fork the project][fork guide] on GitHub.
  - Clone the forked repository to your local development machine
- (e.g. `git clone https://github.com/<YOUR_GITHUB_USER>/flame.git`)
- - Run `flutter pub get` in the cloned repository to get all the dependencies
- - Create a new local branch based on issue number from first step
- (e.g. `git checkout -b 12-new-feature`)
- - Make your changes
- - When committing your changes, make sure to start the commit message with `#<issue-number>`
- (e.g. `git commit -m '#12 - New Feature added'`)
+   (e.g. `git clone git@github.com:<YOUR_GITHUB_USER>/flame.git`).
+
+### Environment Setup
+Flame uses [Melos] to manage the project and dependencies.
+
+To install Melos, run the following command from your terminal:
+
+```bash
+flutter pub global activate melos
+```
+
+Next, at the root of your locally cloned repository bootstrap the projects dependencies:
+
+```bash
+melos bootstrap
+```
+
+The bootstrap command locally links all dependencies within the project without having to
+provide manual [`dependency_overrides`][pubspec doc]. This allows all
+plugins, examples and tests to build from the local clone project.
+
+> You do not need to run `flutter pub get` once bootstrap has been completed.
+
+### Performing changes
+ - Create a new local branch from `main` (e.g. `git checkout -b my-new-feature`)
+ - Make your changes.
+ - When committing your changes, make sure that each commit message is clear
+ (e.g. `git commit -m 'Take in an optional Camera as a parameter to FlameGame'`).
  - Push your new branch to your own fork into the same remote branch
- (e.g. `git push origin 12-new-feature`)
- - On Github goto the
- [pull request page](https://guides.github.com/activities/forking/#making-a-pull-request) on your
- own fork and create a merge request to this repository
-
-# Development Guidelines
- - Documentation should be updated.
- - Examples should be updated.
- - Format the Dart code accordingly.
- - Note the
- [`analysis_options.yaml`](https://github.com/flame-engine/flame/blob/main/analysis_options.yaml)
- and write code as stated in this file
-
-# Test generating of `dartdoc`
- - Run the `./scripts/dartdoc.sh` script to generate the Dart documentation.
- - Output will be placed into `doc/api/`.
+ (e.g. `git push origin my-new-feature`)
```

Comment:
how about s/origin/<your fork> ?
possibly it will not be origin, might make it clearer for people to point it here